core/src/consensus.rs

```diff
@@ -359,6 +429,7 @@ impl Tower {
         None
     }
 
+    // root may be forcibly set by arbitrary replay root slot
```

Comment:
After this pr landed, I'll make `fn root` return `Slot` instead of `Option<Slot>`. I've made certain there must be sensible `root` for every case. And `Option<Slot>` clutters bunch of this `fn`'s callsite...